Code/GraphMol/FMCS/MaximumCommonSubgraph.cpp

```diff
@@ -157,8 +157,8 @@ void MaximumCommonSubgraph::init() {
   labels.clear();
   currentLabelValue = 1;
   nq = QueryMolecule->getNumBonds();
-  QueryBondLabels.resize(nq);
-  for (size_t aj = 0; aj < nq; aj++) {
```

Comment:
I think using  "aj" as a bond iterator was the recipe for disaster here.

 nice catch.